docs/content/2.deploy/providers/firebase.md

```diff
@@ -82,11 +104,94 @@ NITRO_PRESET=firebase yarn build
 firebase emulators:start
 ```
 
-## Deploy to Firebase Hosting via CLI
+## Build and Deploy
 
-Deploying to Firebase Hosting is a simple matter of just running the `firebase deploy` command.
+Deploy to Firebase Hosting by running a nitro build and then running the `firebase deploy` command.
 
 ```bash
 NITRO_PRESET=firebase yarn build
+```
+
+```bash
+npx firebase-tools deploy
+```
+
+If you chose the alternative setup using the Firebase CLI, you can also run:
+
+```bash
 firebase deploy
 ```
+
+## Firebase Functions v2
+
+**Preset:** `firebase-v2` ([switch to this preset](/deploy/#changing-the-deployment-preset))
+
+This preset will deploy to [firebase functions V2](https://firebase.google.com/docs/functions/version-comparison) - the 2nd generation of cloud functions.
+
+It works the same as the [`firebase`](#firebase) preset above, but uses the V2 functions runtime.
+
+### Setup
+
+Follow the instructions for the `firebase` preset above for setup and deploy.
+
+::alert{type="warning"}
+New HTTP and HTTP callable functions deployed with any Firebase CLI lower than version 7.7.0 are private by default and throw HTTP 403 errors when invoked. Either explicitly [make these functions public](https://cloud.google.com/functions/docs/securing/managing-access-iam#allowing_unauthenticated_http_function_invocation) or [update your Firebase CLI](https://firebase.google.com/docs/cli#setup_update_cli) before you deploy. If you deploy the function and it is not publicly accessible, then you will need to visit the [Functions Console](https://console.cloud.google.com/functions/list) on Google cloud platform, click your function name, select the 'Permissions' tab and follow the instructions on how to make the function public.
+::
+
+### Options
+
+The `firebase-v2` preset also supports passing options to the firebase function via the nitro config. These options are defined in the nitro config object under `firebase.v2.httpRequestOptions`.
+
+The `httpRequestOptions` object is passed to `functions.https.onRequest()` function as the first argument as per the [firebase docs](https://firebase.google.com/docs/functions/http-events?gen=2nd#trigger_a_function_with_an_http_request).
+
+For example, to set the region and max instances of the function, you can add the following to your `nitro.config.ts`:
+
+```ts
+import { defineNitroConfig } from 'nitropack/config'
+
+export default defineNitroConfig({
+  // Nitro options...
+  firebase: {
```

Comment:
I'll let @pi0 weigh in on this, as we've already been through a review on it. See the resolved issue here @posva :
https://github.com/unjs/nitro/pull/1142#pullrequestreview-1512458039

Reply:
Just my 2 cents, but for #1377 my idea was to have the config option shared between v1 and v2 functions, so you could seamlessly switch between them without worrying about the config. As someone using this in production, it would be nice to switch between v1 and v2 without worrying about accidentally overwriting function deployments.

Reply:
Could you detail those circumstances? I don't think I see any reason not to upgrade the currently named `server` function that renders the Nuxt app to gen 2 and then rollback to gen 1 🤔 

Publishing a gen 2 function with the same name will error, you have to name it differently

> The main purpose behind sharing the config would be to simply upgrade to v2 without changing config.

Totally 🙌

Reply:
> Publishing a gen 2 function with the same name will error, you have to name it differently

This is good to know, so in that case you would not be able upgrade to v2 with the same function name unless you delete the original first? Perhaps in that case it should actually be a different config value to avoid an error. Maybe even the default name should be different as well then?

As for downgrading to v1, I am imagining a case where we downgrade temporarily either to debug or rollback in case of any bugs that arise after upgrading. Granted, I don't think this scenario is very likely.

Reply:
IMO the name should be different yes.

> so in that case you would not be able upgrade to v2 with the same function name unless you delete the original first? 

Not exactly: upgrading is a bit different as you want to go through a one-time step where you have both gen functions co-existing. In more complex scenarios: https://firebase.google.com/docs/functions/2nd-gen-upgrade.
You could also just redeploy the second gen with a different name and remove the 1st one in one deployment. I'm not completely sure but I imagine you risk losing a fraction of traffic during that deployment, which in some projects is fine, in others isn't.